chain/client/tests/catching_up.rs

```diff
@@ -70,24 +75,45 @@ mod tests {
         WaitingForValidate,
     }
 
+    #[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+    pub struct StateRequestStruct {
+        pub shard_id: u64,
+        pub hash: CryptoHash,
+        pub need_header: bool,
+        pub parts: StateRequestParts,
+        pub target: AccountOrPeerIdOrHash,
+    }
+
     /// Sanity checks that the incoming and outgoing receipts are properly sent and received
     #[test]
     fn test_catchup_receipts_sync_third_epoch() {
-        test_catchup_receipts_sync_common(13, 1)
+        test_catchup_receipts_sync_common(13, 1, false)
+    }
+
+    /// The test aggressively blocks lots of state requests
+    /// and causes at least two timeouts per node (first for header, second for parts).
+    ///
+    /// WARNING! For your convenience, set manually STATE_SYNC_TIMEOUT to 1 before running the test.
```

Comment:
should we make STATE_SYNC_TIMEOUT then a config parameter so in tests can set it to lower value?

Reply:
Yep, we definitely should, but not because to decrease execution time of this test. The test is fine.

There should be separate task to put all important constants into config to add more flexibility to the system.